Library/Homebrew/formulary.rb

```diff
@@ -533,6 +537,13 @@ def self.loader_for(ref, from: nil)
     when URL_START_REGEX
       return FromUrlLoader.new(ref)
     when HOMEBREW_TAP_FORMULA_REGEX
+      # If `homebrew/core` is specified and not installed, check whether the formula is already installed.
+      if ref.start_with?("homebrew/core/") && !CoreTap.instance.installed? && Homebrew::EnvConfig.install_from_api?
+        name = ref.split("/", 3).last
+        possible_keg_formula = Pathname.new("#{HOMEBREW_PREFIX}/opt/#{name}/.brew/#{name}.rb")
```

Comment:
Might be nice to have a helper for this sort of format; it's been reused in a few places.